### ISSUE TITLE
Question about fisheye camera model in cv::fisheye::initUndistortRectifyMap and the document

### ISSUE THREAD

**Issue by Apeiria** [opencv/opencv#23792]
### Describe the doc issue

When I read the source code, I found that the camera model used in `cv::fisheye::initUndistortRectifyMap` is different from what is described in the doc https://docs.opencv.org/4.x/db/d58/group__calib3d__fisheye.html and I'm confused about the difference.

Below are the screenshots of where I think is different.

![image](https://github.com/opencv/opencv/assets/60050293/1f496fdf-cf93-45ae-b143-80013b6596c2)
![image](https://github.com/opencv/opencv/assets/60050293/f3b8f8ea-2cd3-4eb0-81d8-5ef686f25703)

**Comment by Swastyy** [comment]
Hi @asmorkalov and @Apeiria , can I correct this on the doc?

**Comment by Apeiria** [comment]
> Hi @asmorkalov and @Apeiria , can I correct this on the doc?

I have no idea. @asmorkalov  could you confirm whether the document is wrong? Or are the camera models in the code and the document are supposed to be different?

**Comment by asmorkalov** [comment]
@ivashmak could you take a look on it?

**Comment by ivashmak** [comment]
Sure!

**Comment by ivashmak** [comment]
It seems that code and docs are correct for \alpha = 0, in this case we simplify u = f_x x' + c_x, which corresponds to the written code. The parameter \alpha is not defined in OpenCV docs, so it's hard to guess what it means. I've looked at the paper referenced in docs, ie, Kannala-Brandt camera model, and I have not found any reference to \alpha as well, here is the paper if someone wants to have a closer look for understanding \alpha:

https://oulu3dvision.github.io/calibgeneric/Kannala_Brandt_calibration.pdf

**Comment by asmorkalov** [comment]
@ivashmak Could you take a look on the code and document \alpha behaviour.

**Comment by ivashmak** [comment]
1. 
code: `double x = _x/_w, y = _y/_w;`
documentation: $a = x/z$ and $b = y/z$

2.
`double r = sqrt(x*x + y*y);`
$r^2 = a^2 + b^2$

3. 
`double theta = atan(r);`
$\theta = atan(r)$

4.
`double theta2 = theta*theta, theta4 = theta2*theta2, theta6 = theta4*theta2, theta8 = theta4*theta4;`
`double theta_d = theta * (1 + k[0]*theta2 + k[1]*theta4 + k[2]*theta6 + k[3]*theta8);`
$\theta_d = \theta(1 + k_1 \theta^2 + k_2 \theta^4 + k_3 \theta^6 + k_4 \theta^8)$


5.
code:
`double scale = (r == 0) ? 1.0 : theta_d / r;`
`u = f[0]*x*scale + c[0];`
`v = f[1]*y*scale + c[1];`
documentation:
$x^\prime = (\theta_d / r) a$
$y^\prime = (\theta_d / r) b$
$u = f_x  (x^\prime + \alpha y^\prime) + c_x$
$v = f_y  y^\prime + c_y$

Basically, $y^\prime$ = `y*scale` and $x^\prime$ = `x*scale`, $f_x$ = `f[0]`, $f_y$ = `f[1]`, $c_x$ = `c[0]`, $c_y$ = `c[1]`
We open brackets:
$u = f_x (x^\prime + \alpha y^\prime) + c_x = f_x x^\prime + f_x \alpha y^\prime +c_x$
Since $\alpha = 0$ then $u = f_x x^\prime + c_x$ and it matches the code `u = f[0]*x*scale + c[0];`

**Comment by asmorkalov** [comment]
@ivashmak could you create PR?

**Comment by ivashmak** [comment]
Sure, do you want to change the documentation (remove alpha) or change the code (add zero-alpha)?

**Comment by asmorkalov** [comment]
@vpisarev What do you think?
